cmd/otelcol/main.go

```diff
@@ -41,7 +42,9 @@ func main() {
 		GitHash:  version.GitHash,
 	}
 
-	svc, err := service.New(service.Parameters{ApplicationStartInfo: info, Factories: factories})
+	rp := resourcedetection.NewResourceProvider()
```

Comment:
To use custom detectors, a user would need to add the following:

```go
rp.AddDetector(gcpdetector.GCE)
rp.AddDetector(gcpdetector.GKE)
```
_which makes me realize the interface should probably be `AddDetectors(detector ...ResourceDetector)`_